### PR TITLE
feat(auth): throw Error when auth type is not a valid one

### LINKED ISSUE #216: Throw error of unrecognized `type` in `.auth({ type })`
I just spend a significant amount of time debugging a problem that was ultimately caused by a space after `"installation "`:

```js
octokit.auth({ type: "installation " })
```

We should throw an error if a `type` is passed that is unknown. Otherwise it currently falls back to `oauth`, which is not what we want.

### PR DESCRIPTION
**Pull request by oscard0m**
## 📝 Summary

throw Error when auth type is not a valid one

## ⛱ Motivation and Context

Fix #216

### CODE REVIEW

File: test/index.test.ts [hunk 1]
```diff
@@ -58,6 +58,15 @@ test("README example for app auth", async () => {
   });
 });
 
+test("throws if invalid 'type' is provided", async () => {
+  const auth = createAppAuth({
+    appId: APP_ID,
+    privateKey: PRIVATE_KEY,
+  });
+
+  expect(auth({ type: "app2" })).toThrow();
```
Comment (oscard0m) [inline test/index.test.ts:67]: This does not compile:

image

---

image

Comment (gr2m) [reply test/index.test.ts:67]: add a [`@ts-expect-error` comment](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-9.html#-ts-expect-error-comments)

Comment (gr2m) [reply test/index.test.ts:67]:
```suggestion
  // @ts-expect-error TS2322
  expect(auth({ type: "app2" })).toThrow();
```

compare https://github.com/octokit/webhooks.js/search?q=ts-expect-error

**Review by gr2m — APPROVED** [review]
Thanks!

**Pull request merged**